### PR TITLE
Convert controller node files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go [hunk 1]
```diff
@@ -925,17 +925,17 @@ func (nc *Controller) processNoTaintBaseEviction(node *v1.Node, observedReadyCon
...
-				klog.V(2).Infof("Node is unresponsive. Adding Pods on Node %s to eviction queues: %v is later than %v + %v",
-					node.Name,
-					decisionTimestamp,
-					nodeHealthData.readyTransitionTimestamp,
-					nc.podEvictionTimeout-gracePeriod,
+				klog.V(2).InfoS("Node is unresponsive. Adding Pods on Node to eviction queues",
+					"node", node.Name,
+					"timeStamp", decisionTimestamp,
+					"readyTransitionTimestamp", nodeHealthData.readyTransitionTimestamp,
+					"podEvictionTimeout-gracePeriod", nc.podEvictionTimeout-gracePeriod,
```
Comment (bowei) [inline pkg/controller/nodelifecycle/node_lifecycle_controller.go:932]: "difference"

**Review by bowei — COMMENTED** [review]
one comment

File: pkg/controller/nodelifecycle/node_lifecycle_controller.go (hunk 1 above)
Comment (scottilee) [reply pkg/controller/nodelifecycle/node_lifecycle_controller.go:932]: Fixed.

**Comment by scottilee** [comment]
@bowei Thanks for the review! I addressed your comment and made some other minor fixes as well as updated the .structured_logging file.

File: pkg/controller/nodeipam/ipam/range_allocator.go [hunk 2]
```diff
@@ -351,17 +351,17 @@ func (r *rangeAllocator) updateCIDRsAllocation(data nodeReservedCIDRs) error {
...
-			klog.V(4).Infof("Node %v already has allocated CIDR %v. It matches the proposed one.", node.Name, data.allocatedCIDRs)
+			klog.V(4).InfoS("Node already has allocated CIDR. It matches the proposed one.", "node", node.Name, "CIDR", data.allocatedCIDRs)
 			return nil
 		}
 	}
 
 	// node has cidrs, release the reserved
 	if len(node.Spec.PodCIDRs) != 0 {
-		klog.Errorf("Node %v already has a CIDR allocated %v. Releasing the new one.", node.Name, node.Spec.PodCIDRs)
+		klog.ErrorS(nil, "Node already has a CIDR allocated. Releasing the new one.", "node", node.Name, "PodCIDR", "node.Spec.PodCIDRs")
```
Comment (yangjunmyfm192085) [inline pkg/controller/nodeipam/ipam/range_allocator.go:361]:
```suggestion
		klog.ErrorS(nil, "Node already has a CIDR allocated. Releasing the new one.", "node", node.Name, "PodCIDR", node.Spec.PodCIDRs)
```

**Comment by bowei** [comment]
In the future for mechanical changes like these, it is preferable to open many small PRs than a single large PR:

* You will be able to merge them independently
* Reviewers will not need to re-review lines that are already done

File: pkg/controller/nodeipam/ipam/range_allocator.go (hunk 2 above)
Comment (scottilee) [reply pkg/controller/nodeipam/ipam/range_allocator.go]: Fixed. Good catch!

**Comment by scottilee** [comment]
@bowei Noted, thanks for reviewing anyways! I addressed the error and rebased. Let me know if there's anything else I need to fix.

File: pkg/controller/nodelifecycle/scheduler/taint_manager.go [hunk 3]
```diff
@@ -160,7 +160,7 @@ func NewNoExecuteTaintManager(c clientset.Interface, getPod GetPodFunc, getNode
 	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "taint-controller"})
 	eventBroadcaster.StartStructuredLogging(0)
 	if c != nil {
-		klog.V(0).InfoS("Sending events to api server")
+		klog.V(0).InfoS("Sending events to the API Server")
 		eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: c.CoreV1().Events("")})
 	} else {
 		klog.Fatalf("kubeClient is nil when starting NodeController")
```
Comment (yangjunmyfm192085) [inline pkg/controller/nodelifecycle/scheduler/taint_manager.go:166]: klog.Fatalf maybe also need to migrate

Comment (scottilee) [reply pkg/controller/nodelifecycle/scheduler/taint_manager.go]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 I migrated the fatal you pointed out as well as a few others I found. Let me know if there's anything else I missed.

File: pkg/controller/nodelifecycle/scheduler/taint_manager.go [hunk 4]
```diff
@@ -160,10 +161,11 @@ func NewNoExecuteTaintManager(c clientset.Interface, getPod GetPodFunc, getNode
 	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "taint-controller"})
 	eventBroadcaster.StartStructuredLogging(0)
 	if c != nil {
-		klog.V(0).InfoS("Sending events to api server")
+		klog.V(0).InfoS("Sending events to the API Server")
 		eventBroadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: c.CoreV1().Events("")})
 	} else {
-		klog.Fatalf("kubeClient is nil when starting NodeController")
+		klog.ErrorS(nil, "kubeClient is nil when starting NodeController")
```
Comment (yangjunmyfm192085) [inline pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]:
```suggestion
		klog.ErrorS(nil, "KubeClient is nil when starting NodeController")
```

**Comment by yangjunmyfm192085** [comment]
> @yangjunmyfm192085 I migrated the fatal you pointed out as well as a few others I found. Let me know if there's anything else I missed.

Thanks, a little change

File: pkg/controller/nodelifecycle/scheduler/taint_manager.go (hunk 4 above)
Comment (scottilee) [reply pkg/controller/nodelifecycle/scheduler/taint_manager.go:167]: Fixed this one and the others.

**Comment by yangjunmyfm192085** [comment]
part duplicated with https://github.com/kubernetes/kubernetes/pull/100672
/wg structured-logging

**Pull request closed without merging**